src/material/paginator/paginator.ts

```diff
@@ -170,6 +176,15 @@ export class MatPaginator extends _MatPaginatorBase implements OnInit, OnDestroy
   }
   private _showFirstLastButtons = false;
 
+  /** Whether to show the 'All' page size option to the user. */
+  @Input()
+  get showAllPageSizeOption(): boolean { return this._showAllPageSizeOption; }
+  set showAllPageSizeOption(value: boolean) {
```

Comment:
I'm not sure how scalable this will be, because we'd have to keep adding other boolean flags to support other options in the future. Maybe it should be done through the `pageSizeOptions`.

Reply:
I understand.  I will close this PR and continue on #20367.